File: erykandroidcommon/src/main/java/pl/edu/amu/wmi/erykandroidcommon/recycler/EmptyGridDialogFragment.java
package pl.edu.amu.wmi.erykandroidcommon.recycler;

import android.app.AlertDialog;
import android.app.Dialog;
import android.app.DialogFragment;
import android.os.Bundle;
import android.support.annotation.NonNull;

import pl.edu.amu.wmi.erykandroidcommon.R;


public class EmptyGridDialogFragment extends DialogFragment {

    public static EmptyGridDialogFragment getInstance() {
        return new EmptyGridDialogFragment();
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity(), R.style.AppTheme_MyAlertDialogStyle);
        builder
                .setTitle(getString(R.string.no_data))
                .setNegativeButton(android.R.string.ok, (dialogInterface, i) -> {
                    getActivity().finish();
                    dialogInterface.dismiss();
                });

        return builder.create();
    }


}

File: src/main/java/pl/edu/amu/wmi/erykandroidcommon/recycler/EmptyGridDialogFragment.kt
package pl.edu.amu.wmi.erykandroidcommon.recycler

import android.app.AlertDialog
import android.app.Dialog
import android.app.DialogFragment
import android.os.Bundle
import pl.edu.amu.wmi.erykandroidcommon.R

class EmptyGridDialogFragment : DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle): Dialog {
        val builder = AlertDialog.Builder(activity, R.style.Common)
        builder
            .setTitle(getString(R.string.no_data))
            .setNegativeButton(android.R.string.ok) { dialogInterface, _ ->
                activity!!.finish()
                dialogInterface.dismiss()
            }

        return builder.create()
    }

    companion object {
        val instance: EmptyGridDialogFragment get() = EmptyGridDialogFragment()
    }
}
